src/shape/2d_primitives.js

```diff
@@ -79,32 +168,51 @@ define(function (require) {
       height,
       this._ellipseMode
     );
-    var radius = (vals.h > vals.w) ? vals.h / 2 : vals.w / 2,
-      //scale the arc if it is oblong
-      xScale = (vals.h > vals.w) ? vals.w / vals.h : 1,
-      yScale = (vals.h > vals.w) ? 1 : vals.h / vals.w;
-    ctx.save();
-    ctx.scale(xScale, yScale);
+
+    var curves = createArc(1.0, start, stop);
+
     ctx.beginPath();
-    ctx.arc(vals.x, vals.y, radius, start, stop);
-    if (this._doStroke) {
-      ctx.stroke();
-    }
-    if (mode === constants.CHORD || mode === constants.OPEN) {
-      ctx.closePath();
-    } else if (mode === constants.PIE || mode === undefined) {
-      ctx.lineTo(vals.x, vals.y);
-      ctx.closePath();
-    }
+    curves.forEach(function (curve, index) {
+      if (index === 0) {
+        ctx.moveTo(vals.x + curve.x1 * vals.w, vals.y + curve.y1 * vals.h);
+      }
+      ctx.bezierCurveTo(
+        vals.x + curve.x2 * vals.w, vals.y + curve.y2 * vals.h,
+        vals.x + curve.x3 * vals.w, vals.y + curve.y3 * vals.h,
+        vals.x + curve.x4 * vals.w, vals.y + curve.y4 * vals.h
+      );
+    });
+
     if (this._doFill) {
+      if (mode === constants.PIE || mode === undefined) {
+        ctx.lineTo(vals.x, vals.y);
+      }
+      ctx.closePath();
       ctx.fill();
+      if (this._doStroke) {
+        if (mode === constants.CHORD || mode === constants.PIE) {
+          ctx.stroke();
+          return this;
+        }
+      }
     }
-    if(this._doStroke && mode !== constants.OPEN && mode !== undefined) {
-      // final stroke must be after fill so the fill does not
-      // cover part of the line
-      ctx.stroke();
+
+    if (this._doStroke) {
+      if (mode === constants.OPEN || mode === undefined) {
+        ctx.beginPath();
```

Comment:
If the mode is OPEN or undefined (PIE with stroke on the outside only) there is no way to avoid creating a second path.  This always renders the stroke after the fill which is what the original Processing does.